core/client/src/main/java/alluxio/client/block/LocalBlockInStream.java

```diff
@@ -89,14 +90,12 @@ public void close() throws IOException {
       }
       mBlockWorkerClient.unlockBlock(mBlockId);
     } finally {
-      mContext.releaseWorkerClient(mBlockWorkerClient);
+      mClosed = true;
```

Comment:
why do we move this? If `close` throws an exception do we still want to count this as closed?